src/Foundation/Connect/code/Managers/Payment/PaymentManagerV2.cs

```diff
@@ -0,0 +1,75 @@
+﻿//    Copyright 2020 EPAM Systems, Inc.
+// 
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+// 
+//      http://www.apache.org/licenses/LICENSE-2.0
+// 
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+namespace Wooli.Foundation.Connect.Managers.Payment
+{
+    using Base.Models.Logging;
+    using Base.Services.Logging;
+
+    using DependencyInjection;
+
+    using Providers.Contracts;
+
+    using Sitecore.Commerce.Engine.Connect.Entities;
+    using Sitecore.Commerce.Engine.Connect.Pipelines.Arguments;
+    using Sitecore.Commerce.Entities.Carts;
+    using Sitecore.Commerce.Entities.Payments;
+    using Sitecore.Commerce.Services;
+    using Sitecore.Commerce.Services.Payments;
+    using Sitecore.Diagnostics;
+
+    using GetPaymentMethodsRequest = Sitecore.Commerce.Engine.Connect.Services.Payments.GetPaymentMethodsRequest;
+
+    [Service(typeof(IPaymentManagerV2), Lifetime = Lifetime.Singleton)]
+    public class PaymentManagerV2 : BaseManager, IPaymentManagerV2
+    {
+        private readonly PaymentServiceProvider paymentServiceProvider;
+
+        public PaymentManagerV2(IConnectServiceProvider connectServiceProvider, ILogService<CommonLog> logService) : base(logService)
+        {
+            Assert.ArgumentNotNull(connectServiceProvider, nameof(connectServiceProvider));
+
+            this.paymentServiceProvider = connectServiceProvider.GetPaymentServiceProvider();
+        }
+
+        public PaymentClientTokenResult GetPaymentClientToken()
+        {
+            return this.Execute(
```

Comment:
Is it possible to do it like other methods? with request and response.

Reply:
I haven't found such provider, we can do it by extending `PaymentServiceProvider`.

Reply:
`commerce.payments.getClientToken` pipeline is specific for `Sitecore.Commerce.Engine` and doesn't come with Sitecore.Commerce.Connect